File: CodeForces/CodeTON Round 1 (Div. 1 + Div. 2, Rated, Prizes!)/B.cpp
#include <bits/stdc++.h>
using namespace std;
#define MOD 1000000007
typedef long long ll;

ll n;

void solve(){
    ll k, res = 0, i;
    cin>>n>>k;
    unordered_set<ll> s;
    while(n--){
        cin>>i;
        if(s.count(i-k) or s.count(i+k)) res = 1;
        s.insert(i);
    }
    if(res) cout<<"YES"<<endl;
    else cout<<"NO"<<endl;
}

int main(){

    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    ll t;
    cin>>t;
    while(t--){
        solve();
    }
    return 0;
}

File: CodeForces/CodeTON Round 1 (Div. 1 + Div. 2, Rated, Prizes!)/B.py
t = int(input())
while t:
    t-=1
    res = 0
    l = list(map(int,input().split())) 
    n = l[0]
    k = l[1]
    s = set()
    l = list(map(int,input().split())) 
    for i in l:
        if i+k in s or i-k in s:
            res = 1
        s.add(i)
    if res:
        print("YES")
    else:
        print("NO")